GeneratorInterface/LHEInterface/plugins/ExternalLHEProducer.cc

```diff
@@ -330,10 +331,16 @@ void ExternalLHEProducer::beginRun(edm::Run const& run, edm::EventSetup const& e
 void ExternalLHEProducer::endRun(edm::Run const& run, edm::EventSetup const& es) {
   nextEvent();
   if (partonLevel_) {
-    throw edm::Exception(edm::errors::EventGenerationFailure)
-        << "Error in ExternalLHEProducer::endRunProduce().  "
-        << "Event loop is over, but there are still lhe events to process."
-        << "This could happen if lhe file contains more events than requested.  This is never expected to happen.";
+    // VALIDATION_RUN env variable allows to finish event processing early without errors by sending SIGINT
+    if (getenv("VALIDATION_RUN") != nullptr) {
```

Comment:
Please change to `std::getenv()` as `::getenv()` is not thread safe, see https://github.com/cms-sw/cmssw/issues/28073.

But wouldn't an untracked configuration parameter do the job as well?

Reply:
Sure, I'll change the `getenv()`.
Could you elaborate about untracked configuration parameter, I am not sure I understand it?

Reply:
In general we'd want to minimize the use of environment variables to control the job's behavior, and use configuration parameters instead. E.g. here there might be other special use cases to disable the exception.

Reply:
Would such "switch" (boolean) be one of `ExternalLHEProducer` parameters?  If yes, how could this be switched from True to False without manual edit? Let me explain what I mean by that: we start with users giving us a "fragment" i.e. some piece of python code that goes as the first argument for the cmsDriver.py and it may look something like
```python
...
externalLHEProducer = cms.EDProducer("ExternalLHEProducer",
    args = cms.vstring('...'),
    nEvents = cms.untracked.uint32(5000),
    numberOfParameters = cms.uint32(1),
    outputFile = cms.string('cmsgrid_final.lhe'),
    scriptName = cms.FileInPath('...')
)
...
```
We don't manually check or edit it, just use this "fragment" to build two configs: first one we are going to test ("validate" in 8h HTCondor job) and the other one will be submitted for the main production. We want to suppress the exception in the validation, but not in production environment.
Adding (well, _asking all of our users to add_) the "switch" as one of these arguments to their fragments, e.g. `validationRun=cms.untracked.bool(os.getenv('VALIDATION_RUN'))` (this will also require an `import os`) will take years of responding to emails "Why my validations are failing?" until everyone gets it. The underlying goal is to take away as much user input as possible as we are currently overwhelmed with user complaints about failing validations.
Maybe this can somehow be solved by adding another parameter to the `ConfigBuilder.py` that would do the switching, but I am not sure how this should be done, if it is possible at all, to change argument list for the `cms.EDProducer`?